components/in_memory_engine/src/write_batch.rs

```diff
@@ -472,11 +485,24 @@ impl WriteBatch for RegionCacheWriteBatch {
     }
 
     fn prepare_for_region(&mut self, region: &metapb::Region) {
+        let cached_region = CacheRegion::from_region(region);
+        if let Some(current_region) = &self.current_region
+            && current_region == &cached_region
```

Comment:
Is it possible that epoch version changes between this two prepare? I think maybe check the region id is enough

Reply:
Good catch